indykite/import_utils.go

```diff
@@ -35,9 +36,36 @@ func basicStateImporter(_ context.Context, data *schema.ResourceData, _ any) ([]
 }
 
 func parseImportID(d *schema.ResourceData) error {
-	if gidBase64Regex.MatchString(d.Id()) {
+	importID := d.Id()
+
+	// Check if it's a GID (direct ID import)
+	if gidBase64Regex.MatchString(importID) {
 		return nil
 	}
 
-	return errors.New("Unimplemented id format: " + d.Id())
+	// Check if it's a name with location parameter (e.g., "my-resource?location=gid:xxx")
+	// The ID will be kept as-is and used in the read function
+	if strings.Contains(importID, "?location=") {
+		return nil
+	}
+
+	return errors.New("Unimplemented id format: " + importID +
+		". Expected either 'gid:xxx' or 'resource-name?location=gid:xxx'")
+}
+
+// buildReadPath constructs the API path for reading a resource.
+// It supports both:
+// - Direct ID: data.Id() = "gid:xxx" -> returns "/resource/gid:xxx".
+// - Name with location: data.Id() = "my-name?location=gid:xxx" -> returns "/resource/my-name?location=gid:xxx"..
+func buildReadPath(resourcePath string, data *schema.ResourceData) string {
+	id := data.Id()
+
+	// If the ID contains a query parameter, it's a name+location format
+	// Just append it directly to the resource path
+	if strings.Contains(id, "?") {
+		return resourcePath + "/" + id
+	}
+
+	// Otherwise, it's a direct ID
+	return resourcePath + "/" + id
```

Comment:
Issue: This seems as bug. Path inside if and outside if are exactly the same.

Reply:
updated